cpp/tests/utilities/column_wrapper.hpp

```diff
@@ -818,5 +842,276 @@ class lists_column_wrapper : public detail::column_wrapper {
   bool root = false;
 };
 
+namespace detail {
+/**
+ * @brief Convert between arithmetic and chrono types where possible.
+ **/
+template <typename SourceT, typename TargetT, typename InputIterator, typename ValidityIterator>
```

Comment:
This is a lot of new code and complexity. I'd like to find a better solution. 

So to be clear, the problem you're trying to solve is:
```
fixed_width_column_wrapper<cudf::duration_S> w{ 1, 2, 3, 4};
```

This won't currently compile because `duration_S` isn't constructible from an `int`?

Reply:
duration types aren't implicitly convertible from these values and they are used in typed tests. the description is [here](https://github.com/sriramch/cudf/blob/unary_binary/cpp/tests/utilities/column_wrapper.hpp#L977). further, when the constructor to create timestamp using tick counts is removed, we would want the timestamps to be created as well with these values so that they can continue to be used in the typed tests.

Reply:
So I would revisit an idea that @codereport originally had here: https://github.com/rapidsai/cudf/pull/3581/files/182e0a03787e8884a91607d35cea59f99ea7be2f#r400389286

You add a second template arg to `fixed_width_column_wrapper` that is normally defaulted to be the same as the first arg, and it tells you the type you're converting _from_. The first arg is the type you're converting _too_, and you just always do this conversion. When To and From are the same, no conversion happens.

So for duration:

```
fixed_width_column_wrapper<cudf::duration_S, int32_t> w{ 1, 2, 3, 4};
```

Reply:
i'll take a look at the referenced link tomorrow. i don't think i'm clear on the semantics on the 2nd type to the wrapper? is this to facilitate type deduction of the values present in the initializer list? i'm thinking even with this approach we would need something like the fixed width type converter that is present in the pr. further, the column wrappers could be created with iterators as well. in that case, what would be the 2nd type to the template?

this approach was chosen because i thought you didn't want the fixed width column wrapper to do something behind the callers back and want it to simply use the values the caller provided to create the column. hence, the new set of api's were added so that callers consciously choose them, if they need the conversions to be applied.

Reply:
>  the column wrappers could be created with iterators as well. in that case, what would be the 2nd type to the template?

The `value_type` of the iterator. 

> this approach was chosen because i thought you didn't want the fixed width column wrapper to do something behind the callers back and want it to simply use the values the caller provided to create the column. hence, the new set of api's were added so that callers consciously choose them, if they need the conversions to be applied.

Yes, but it's a lot of new code that's redundant with `fixed_width_column_wrapper`. Having the second template argument is a way to make the caller opt-in to the conversion as opposed to having it done implicitly.

Reply:
please look at e5d55b9. i have left the factory methods for now to get the builds to pass. i'll remove them if this looks ok.